tensorflow/core/ops/nn_ops.cc

```diff
@@ -1979,6 +1979,49 @@ precision: Computed Precision at `k` as a `bool Tensor`.
 
 )doc");
 
+// This is the same as `InTopK`, but takes `k` as in input rather than an attr.
+REGISTER_OP("InTopKV2")
+    .Input("predictions: float")
+    .Input("targets: T")
+    .Input("k: T")
+    .Output("precision: bool")
+    .Attr("T: {int32, int64} = DT_INT32")
```

Comment:
Thanks @drpngx for the review. I am still trying to be familiar with the codebase of tensorflow., though it looks like `InTopK` and `InTopKV2` kernel are only registered with CPU. Not sure if  `HostMemory` is needed?